src/dehtml.rs

```diff
@@ -172,15 +230,52 @@ fn dehtml_starttag_cb<B: std::io::BufRead>(
             }
         }
         "b" | "strong" => {
-            dehtml.strbuilder += "*";
+            if dehtml.get_add_text() != AddText::No {
+                dehtml.strbuilder += "*";
+            }
         }
         "i" | "em" => {
-            dehtml.strbuilder += "_";
+            if dehtml.get_add_text() != AddText::No {
+                dehtml.strbuilder += "_";
+            }
         }
         _ => {}
     }
 }
 
+fn pop_tag(count: &mut Option<i32>) {
+    if let Some(ref mut divs) = count {
+        *divs -= 1;
+        if *divs <= 0 {
```

Comment:
Maybe replace counter with `Option<i32>` and check this condition before decreasing this number? Just to make `Some(-1)` not representable.

Reply:
I decided against this as I remember debugging a panic where the cause turned out to be decreasing an `u32` below 0. But panics on Android are pretty hard to debug as you don't get any meaningful info (and `ndk-stack` did add some extra lines but they didn't say anything about the actual cause, either).

So, I'd say that panics are both worse for users and more difficult to debug than logic errors.

But maybe there is a Clippy option to deny subtracting from an unsigned value (there is `checked_sub()` for this) and I just couldn't find it?

Reply:
I think in release mode integers simply wrap around without causing panic. Recent panic fixed by #1803 was related to time durations, not integers.

You can deny [`integer_arithmetic`](https://rust-lang.github.io/rust-clippy/master/index.html#integer_arithmetic) in some functions or the whole module if you want. I would not try to enable it for whole repository though, refactoring large amount of code can make things worse.